addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,705 @@
+            SLEEP_LONG_TIME = 3000;
+            doCrashScan = true;
+            doAllDataScan = true;
+            doTimedScan = true;
+            doJsonScan = true;
+            getMoreConfidence = true;
+            doAuthBypass = true;
+        }
+    }
+
+    @Override
+    public void scan(HttpMessage msg, NameValuePair originalParam) {
+        // TODO add TYPE_JSON control as soon as available
+        isJsonPayload = originalParam.getType() == NameValuePair.TYPE_POST_DATA;
+        // & originalParam.getType() == NameValuePair.TYPE_JSON;
+        super.scan(msg, originalParam);
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        boolean isBingo = false;
+        List<Integer> rtt = new ArrayList<>();
+        HttpMessage msgInjAttack, counterProofMsg;
+        String bodyInjAttack,
+                bodyCounterProof,
+                bodyBase = getBaseMsg().getResponseBody().toString();
+
+        if (!inScope(getTech())) {
+            return;
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(
+                    "\nScannning URL ["
+                            + msg.getRequestHeader().getMethod()
+                            + "] ["
+                            + msg.getRequestHeader().getURI()
+                            + "] on param: ["
+                            + param
+                            + "] with value: ["
+                            + value
+                            + "] for MongoDB Injection");
+        }
+        // injection attack to url-encoded query parameters
+        if (doAllDataScan) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(
+                        "\nStarting with the url-encoded query injection package of attack rules:");
+            }
+            int index = 0;
+            for (String valueInj : ALL_DATA_VALUE_INJECTION) {
+                String paramInj = param + ALL_DATA_PARAM_INJECTION[index++];
+                if (isStop()) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(STOP_LOG);
+                    }
+                    return;
+                }
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("\nTrying with the value: " + paramInj + valueInj);
+                }
+                try {
+                    msgInjAttack = getNewMsg();
+                    setParameter(msgInjAttack, paramInj, valueInj);
+                    sendAndReceive(msgInjAttack, false);
+                    rtt.add(msgInjAttack.getTimeElapsedMillis());
+                    bodyInjAttack = msgInjAttack.getResponseBody().toString();
+                    if (msgInjAttack.getResponseHeader().getStatusCode() != HttpStatusCode.OK) {
+                        continue;
+                    }
+                    if (!bodyBase.equals(bodyInjAttack)) {
+                        // Get more confidence
+                        if (getMoreConfidence) {
```

Comment:
Hello, I think this will raise a lot of false positives because you just need some dynamic content in the response or a input reflection to make the ```!bodyBase.equals(bodyInjAttack)``` condition true. I think that 1 more request is worth it to avoid false positives. This request will be made just in the false positive cases, so the impact is even less. I would remove the ```if (getMoreConfidence``` condition. Thx

Reply:
Hello @DiogoMRSilva. Yes, the false positive may be a problem when a counter-proof is not active (that is only in the LOW streinght configuration). 
I know this control can be improved replacing the simple text comparison with some more sophisticated control  but that still represents a beginning.

Anyway, as you suggest, I could remove the condition of the counter-proof and always execute it.

Thank you so much :)